### ISSUE TITLE
When having Nameservers accross multiple AS's, it's still complaining

### ISSUE THREAD

**Issue by altf4arnold** [zonemaster/zonemaster#1267]
I expect the system to tell that I have a problem with nameservers on same prefix and same ASN, only when those are the only servers (becoming a SPOF).

However, we have multiple providers that all have different ASN's and prefixes. Yet, it still see's it as a problem

**Comment by matsduf** [comment]
Thank you for reporting. If you refer to the same problem as reported in https://github.com/zonemaster/asnlookup-for-zonemaster/issues/1 then we have just become aware of it an a fix of it is under way.

**Comment by matsduf** [comment]
Is it possible for you to switch language to English in Zonemaster and then make a new screen shot? My French is poor. Or provide me the domain name, here or in an email to me (see my profile)?

**Comment by altf4arnold** [comment]
Sorry for that (was in the train so replied fast).

Here is a screenshot in French.
![Screenshot 2024-05-22 at 00 30 01](https://github.com/zonemaster/zonemaster/assets/23227801/e520afd9-0f5b-48e8-a800-64762aa553b0)

Like I said, the information displayed is indeed correct. But the way it is displayed makes it look like a problem. In this situation, there are other servers available on other providers. Which means that there is in fact prefix diversity

**Comment by matsduf** [comment]
This is not related to the issue I referred to. This is related to issue https://github.com/zonemaster/zonemaster/issues/1174

In the case of IPv6 you seem to have diversity, which I think should not cause a WARNING. In the case of IPv4 all NS are on the same prefix. Isn't a WARNING reasonable there?

**Comment by altf4arnold** [comment]
It is not. Because not all NS's are on the same IPv4 Prefix.
Our DNS's are like this : 

2 DNS on AS16276
1 DNS on AS24940
5 DNS on AS6939

We have diversity. But if for example I take down 5 servers and keep one in every ASN, the warning will go away.

The warning seems to come as soon as more than 1 server are in the same ASN or Prefix. Not considering the possibility that those might not be the only existing servers

**Comment by bleve** [comment]
Exactly. Thre is quite serious logic issue here.  Logics should be like there is with ASN. Only if all servers are in same prefix there should be complaint. Now three is always warning if two of the servers are in same prefix. So when you have two servers in different prefixes, check is ok. But when you add third server, in one of the prefixes, warning is added while already existing prefix distribution should remove that.

**Comment by matsduf** [comment]
@altf4arnold, test case Connectivity04 does not look at the ASN. It looks at the prefix. The screen shot in https://github.com/zonemaster/zonemaster/issues/1267#issuecomment-2123541074 says that all IPv4 addresses are in the same prefix.

Test case Connectivity03 is about ASN.

@bleve, please open a new issue on the ASN check in Connectivity03 with your comments.
